### PR TITLE
Tweak: Support Dynamic Number

### PR DESCRIPTION
**Pull request by nuritsha**

### CODE REVIEW

File: includes/controls/select.php [hunk 1]
```diff
@@ -44,6 +46,9 @@ public function get_type() {
 	protected function get_default_settings() {
 		return [
 			'options' => [],
+			'dynamic' => [
+				'categories' => [ TagsModule::NUMBER_CATEGORY ],
```
Comment (matipojo) [inline includes/controls/select.php]: Why only numbers?

File: includes/controls/slider.php [hunk 2]
```diff
@@ -90,29 +96,31 @@ public function content_template() {
...
+						<div class="elementor-slider__scales">
+							<# for ( var i = 0; i < data.scales; i++ ) { #>
+								<div class="elementor-slider__scale"></div>
+							<# } #>
+						</div>
 						<# } #>
 					</div>
 				<# } #>
-				<div class="elementor-slider"></div>
+				<div class="elementor-slider "></div>
```
Comment (matipojo) [inline includes/controls/slider.php]: Remove space?

File: includes/widgets/progress.php [hunk 3]
```diff
@@ -350,12 +359,14 @@ protected function render() {
 	protected function _content_template() {
 		?>
 		<#
+		const progress_percentage = 100 < settings.percent.size ? 100 : settings.percent.size;
```
Comment (matipojo) [inline includes/widgets/progress.php]: Check if it's numeric.

File: includes/widgets/progress.php [hunk 4]
```diff
@@ -371,8 +382,9 @@ protected function _content_template() {
 		<div {{{ view.getRenderAttributeString( 'progressWrapper' ) }}}>
 			<div class="elementor-progress-bar" data-max="{{ settings.percent.size }}">
 				<span {{{ view.getRenderAttributeString( 'inner_text' ) }}}>{{{ settings.inner_text }}}</span>
-				<# if ( 'hide' !== settings.display_percentage ) { #>
-					<span class="elementor-progress-percentage">{{{ settings.percent.size }}}%</span>
+				<#
+				if ( 'hide' !== settings.display_percentage ) { #>
```
Comment (matipojo) [inline includes/widgets/progress.php]: Remove line break.

File: includes/widgets/star-rating.php [hunk 5]
```diff
@@ -329,11 +332,11 @@ protected function get_rating() {
 	 */
 	protected function render_stars( $icon ) {
 		$rating_data = $this->get_rating();
-		$rating = $rating_data[0];
-		$floored_rating = (int) $rating;
+		$rating = (float) $rating_data[0];
```
Comment (matipojo) [inline includes/widgets/star-rating.php]: Why? (add docs)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/controls/select.php (hunk 1 above)
Comment (nuritsha) [reply includes/controls/select.php]: removed from scope of this PR

File: includes/controls/slider.php (hunk 2 above)
Comment (nuritsha) [reply includes/controls/slider.php]: done

File: includes/widgets/progress.php (hunk 4 above)
Comment (nuritsha) [reply includes/widgets/progress.php]: done

File: includes/widgets/star-rating.php (hunk 5 above)
Comment (nuritsha) [reply includes/widgets/star-rating.php]: done

File: includes/widgets/progress.php [hunk 6]
```diff
@@ -350,12 +359,17 @@ protected function render() {
 	protected function _content_template() {
 		?>
 		<#
+		let progress_percentage = 0;
+		if ( ! isNaN() ) {
```
Comment (matipojo) [inline includes/widgets/progress.php]: Did you mean `isNaN( settings.percent.size )`?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/widgets/progress.php (hunk 6 above)
Comment (nuritsha) [reply includes/widgets/progress.php]: walla... done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**